psi-sample/host/src/main/java/com/r3/conclave/sample/host/Host.java

```diff
@@ -0,0 +1,91 @@
+package com.r3.conclave.sample.host;
+
+import com.r3.conclave.host.AttestationParameters;
+import com.r3.conclave.host.EnclaveHost;
+import com.r3.conclave.host.EnclaveLoadException;
+import com.r3.conclave.host.MailCommand;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Host accepts client connections, and relays messages from clients to enclaves. It also relays replies back from enclave to clients.
+ */
+public class Host {
+
+    private static Map<String, Socket> clientMap = new HashMap<>();
+
+    public static void main(String args[]) throws EnclaveLoadException, IOException {
+        final String enclaveClassName = "com.r3.conclave.sample.enclave.ReverseEnclave";
+
+        //Check if Platform supports enclave.
+        try {
+            EnclaveHost.checkPlatformSupportsEnclaves(true);
+        } catch (EnclaveLoadException e) {
+            e.printStackTrace();
+        }
+
+        //loads enclave
+        EnclaveHost enclaveHost = EnclaveHost.load(enclaveClassName);
+
+        //Open a TCP server socket for any client who wishes to connect to the host
+        ServerSocket acceptor = new ServerSocket(9999);
+        System.out.println("Listening on port 9999");
+
+        while (true) {
+            //accept is a blocking call and will wait for any client to connect
+            Socket clientSocket = acceptor.accept();
+
+            //routing hint identifies a particular client. This is used by enclave to send reply back to host
+            String routingHint = UUID.randomUUID().toString();
```

Comment:
If we do decide to keep the client type in the unencrypted envelope, we might as well use it for the routing hint rather than using a random UUID

Reply:
It is a good idea to use client type as a routing hint if we would have kept client type in the envelope.